AlertaDengue/api/views.py

```diff
@@ -168,15 +175,23 @@ def get(self, request):
                     'The output format available are: `csv` or `json`.'
                 )
 
-            eyw_start = ey_start * 100 + ew_start
-            eyw_end = ey_end * 100 + ew_end
+            # EpidemEpidemiological Week default format = YYYYWW
+            epi_yearweek_start = epiyear_start * 100 + epiweek_start
+            epi_yearweek_end = epiyear_end * 100 + epiweek_end
+
+            df = None
+
+            while True:
```

Comment:
*Pass this code block to a specific query directly in the base and avoid too many requests*

Reply:
Agreed, I am not sure what the purpose of this query is, but it is requesting the same data over and over, only moving the starting week further into the past 3 weeks at a time, until it returns a non-empty dataframe. It is much easier to simply do a 
```sql
select geocode,disease, ew_start, ew_end from alertcity order by ew_start desc limit 3;
```

Reply:
> Agreed, I am not sure what the purpose of this query is, but **it is requesting the same data over and over**, only moving the starting week further into the past 3 weeks at a time, until it returns a non-empty dataframe. It is much easier to simply do a
> 
> ```sql
> select geocode,disease, ew_start, ew_end from alertcity order by ew_start desc limit 3;
> ```
Using the limit clause does not fetch the last three weeks with the non-empty data frame and still limits the API data query to only 3 records for queries greater than three weeks.

Reply:
I am passing to **ibis expr** the *[AlertCity.search](https://github.com/AlertaDengue/AlertaDengue/blob/main/AlertaDengue/api/db.py#L557)* query in db.py 

``` 
...
...
...
def search_ibis(
        disease: str, 
        geocode: int, 
        ew_start: Optional[int] = None, 
        ew_end: Optional[int] = None

) -> ibis.expr:
        """
        Return an ibis expression with the history for a given geocode by disease.
        Parameters
        ----------
        disease : str, {'dengue', 'chik', 'zika'}
        geocodes : List[int]
        ew_start : Optional[int]
            The starting Year/Week, e.g.: 202202
        ew_end : Optional[int]
            The ending Year/Week, e.g.: 202205
        Returns
        -------
        ibis.expr.types.Expr
        """
        if disease not in CID10.keys():
            raise Exception(
                'The diseases available are: %s.'
                % ', '.join('`%s`' % k for k in CID10.keys())
            )

        table_suffix = ''
        if disease != 'dengue':
            table_suffix = F'_{disease}'

        schema_city = con.schema('Municipio')
        t_hist = schema_city.table('Historico_alerta{}'.format(table_suffix))

        if ew_start and ew_end :
            print('Use the selected epiweek...')

        else:
            ew_end = t_hist['SE'].max().execute()
            ew_start = ew_end - 2
            print(F'Required missing parameters; return the last three epiweek {ew_start} --> {ew_end}')

        hist_filter = (
            t_hist['SE'].between(int(ew_start), int(ew_end))
        ) & (t_hist['municipio_geocodigo'] == geocode )

        return t_hist[hist_filter].execute()
```